src/test/regress/sql/row_types.sql

```diff
@@ -0,0 +1,43 @@
+-- Tests for row types on the target list
+CREATE SCHEMA row_types;
+SET search_path TO row_types;
+
+CREATE TABLE test (x int, y int);
+SELECT create_distributed_table('test','x');
+
+CREATE OR REPLACE FUNCTION table_returner(INT) RETURNS TABLE(name text, id INT)
+AS $$
+BEGIN
+   RETURN QUERY SELECT $1::text, $1;
+END;
+$$ language plpgsql;
+SELECT create_distributed_function('table_returner(int)');
+
+CREATE OR REPLACE FUNCTION record_returner(INOUT id int, OUT name text)
+RETURNS record
+AS $$
+BEGIN
+   id := id + 1;
+   name := 'returned';
+END;
+$$ language plpgsql;
+SELECT create_distributed_function('record_returner(int)');
```

Comment:
Until we merge specifying distribution argument, I've been manually updating pg_dist_object in multi_mx_call test

Reply:
There's no distribution argument here, I just used create_distributed_function to be able to call the function from a query.